components/Input/index.tsx

```diff
@@ -0,0 +1,31 @@
+import { ComponentPropsWithoutRef, forwardRef } from "react";
+
+import classNames from "classnames/bind";
+
+import styles from "./Input.module.scss";
+
+const cn = classNames.bind(styles);
+
+interface InputProps extends ComponentPropsWithoutRef<"input"> {
+  label: string;
+  errorMessage?: string;
+  isRequired?: boolean;
+  isOptional?: boolean;
+}
+
+export default forwardRef<HTMLInputElement, InputProps>(function Input(
+  { label, id, errorMessage, isRequired, isOptional, ...restProps }: InputProps,
+  ref,
+) {
+  return (
+    <div className={cn("container")}>
+      <label htmlFor={id}>
+        {label}
+        {isRequired && <span className={cn("required-field")}>*</span>}
+        {isOptional && <span className={cn("selective-field")}>(선택)</span>}
```

Comment:
```suggestion
        {props.required ? // isRequired를 새로 만들지 않고 input 기본 속성인 required 사용
         <span className={cn("required-field")}>*</span> 
         : 
         <span className={cn("selective-field")}>(선택)</span>}
```
p4: 인풋창 입력할 때 필수 or 선택인 것 같은데 `isRequired` 하나만 있어도 삼항 연산자로 가능할 것 같아요
그리고 `isRequired`를 새로 만드는 것보다 `input` 기본 속성에 있는 `required` 을 사용하는게 웹 접근성도 좋을 것 같은데 어떻게 생각하시나요??

Reply:
```tsx
{restProps.required && <span className={cn("required-field")}>*</span>}
```

지난 번에 말씀주셨던 대로 위와 같이 `required` 속성을 사용해보려고 했는데, 이렇게 코드를 작성했더니 `required` 값이 `undefined`로 넘어오길래, 따로 속성을 지정하여 넘겨주는 방식을 사용하였습니다.

`isRequired`와 `isOptional`을 분리시킨 이유는, `isRequired` 값이 `false`인게 선택 입력 input이라는 게 다소 이상하게 느껴져서, 조금 더 명확하게 표현을 해보고자 `isOptional`을 사용하였습니다.

`isRequired` 하나로만 작성하는 게 더 나을까요?

Reply:
방금 `restProps.required` 이렇게 해봤는데 `true`로 잘 넘어오는 것 같아요!
`isRequired` 값이 `false` 일 때는 선택 입력 `input`인게 더 자연스럽지 않나요...?
필수, 선택 이외의 옵션 더 있으면 남주님처럼 나눠주는게 더 좋은 것 같은데 필수 or 선택 2가지 조건이면 삼항연산자가 더 좋은 것 같아요(제 개인적인 의견...)
![스크린샷 2024-06-06 오후 3 13 26](https://github.com/tickita-project/tickita-front/assets/154664697/6faa21e5-e094-4b35-8b20-ee484a97b5fe)

Reply:
저는 왜 계속 `undefined`로 넘어올까여....😭

우선은 `isOptional` 제거하고 `isRequired`만 사용하는 걸로 코드 수정하겠습니다!!

`required` 사용하는 부분은 조금 더 찾아보겠습니다 😢